### PR TITLE
update roadmap for 3.4

### LINKED ISSUE #482: Update openssl roadmap.
Thank you for releasing openssl 3.3.0 
And thank you for your hard work!
I would like to know what is planned next, for openssl 3.4.0? or maybe farther. 
Can you update the roadmap? 

https://www.openssl.org/roadmap.html

### PR DESCRIPTION
**Pull request by nhorman**
Add in epic items that we have planned for 3.4 and reference the running 3.4 schedule

Fixes #24219

### CODE REVIEW

File: roadmap.md [hunk 1]
```diff
@@ -12,6 +12,20 @@ We encourage our community to follow the project's progress and participate in
 its evolution. Stay updated and engage with us through our [project board],
 where you can track real-time updates, milestones, and changes.
 
+### OpenSSL 3.4
+OpenSSL 3.4 is targeting the following enhancements:
+* Expanded use of RCU locks where appropriate for performance enhancements
+* Optimization of of algorithm fetching to improve performance
```
Comment (t8m) [inline roadmap.md]: Duplicate `of`

File: roadmap.md [hunk 2]
```diff
@@ -12,6 +12,20 @@ We encourage our community to follow the project's progress and participate in
...
+* Composite algorithm implementations
+* Implementation of Attribute Certificates
+* Elimination of the use of atexit
+* DTLS 1.3 support
```
Comment (t8m) [inline roadmap.md]: This is unlikely to make 3.4.

File: roadmap.md [hunk 3]
```diff
@@ -12,6 +12,19 @@ We encourage our community to follow the project's progress and participate in
...
 where you can track real-time updates, milestones, and changes.
 
+### OpenSSL 3.4
+OpenSSL 3.4 is targeting the following enhancements:
+* Expanded use of RCU locks where appropriate for performance enhancements
+* Optimization of algorithm fetching to improve performance
+* Composite algorithm implementations
+* Implementation of Attribute Certificates
+* Elimination of the use of atexit
+* QUIC Unreliable datagram extensions (RFC 9221)
```
Comment (hlandau) [inline roadmap.md]:
```suggestion
* Server-side QUIC support
```

We are not shipping RFC 9221 for 3.4...

**Review by hlandau — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Comment by t8m** [comment]
Merged. Thank you.

**Pull request closed without merging**